internal/testing/e2e/backfill_test.go

```diff
@@ -66,3 +68,134 @@ func TestCreateGetBackfill(t *testing.T) {
+			SearchFields: &pb.SearchFields{
+				StringArgs: map[string]string{
+					"field": "value",
+				},
+			},
+		},
+	})
+	require.Nil(t, err)
+
+	b := &pb.Backfill{
+		SearchFields: &pb.SearchFields{
+			StringArgs: map[string]string{
+				"field": "value",
+			},
+		},
+	}
+	m := &pb.Match{
+		MatchId:  "1",
+		Tickets:  []*pb.Ticket{t1},
+		Backfill: b,
+	}
+
+	om.SetMMF(func(ctx context.Context, profile *pb.MatchProfile, out chan<- *pb.Match) error {
+		out <- m
+		return nil
+	})
+
+	om.SetEvaluator(func(ctx context.Context, in <-chan *pb.Match, out chan<- string) error {
+		p, ok := <-in
+		require.True(t, ok)
+
+		out <- p.MatchId
+		return nil
+	})
+
+	stream, err := om.Backend().FetchMatches(ctx, &pb.FetchMatchesRequest{
+		Config:  om.MMFConfigGRPC(),
+		Profile: &pb.MatchProfile{},
+	})
+	require.Nil(t, err)
+	resp, err := stream.Recv()
+	require.Nil(t, err)
+
+	actual, err := om.Frontend().GetBackfill(ctx, &pb.GetBackfillRequest{BackfillId: resp.Match.Backfill.Id})
+	require.Nil(t, err)
+	require.NotNil(t, actual)
+
+	b.Id = actual.Id
+	b.CreateTime = actual.CreateTime
+	require.True(t, proto.Equal(b, actual))
+	
+	client, err := om.Query().QueryTickets(ctx, &pb.QueryTicketsRequest{Pool: &pb.Pool{
+		StringEqualsFilters: []*pb.StringEqualsFilter{{StringArg: "field", Value: "value"}},
+	}})
+
+	require.Nil(t, err)
+	_, err = client.Recv()
+	require.Equal(t, io.EOF, err)
+}
+
+func TestProposedBackfillUpdate(t *testing.T) {
+	ctx := context.Background()
+	om := newOM(t)
+	t1, err := om.Frontend().CreateTicket(ctx, &pb.CreateTicketRequest{
+		Ticket: &pb.Ticket{
+			SearchFields: &pb.SearchFields{
+				StringArgs: map[string]string{
+					"field": "value",
+				},
+			},
+		},
+	})
+	require.Nil(t, err)
+
+	b, err := om.Frontend().CreateBackfill(ctx, &pb.CreateBackfillRequest{Backfill: &pb.Backfill{
+		SearchFields: &pb.SearchFields{
+			StringArgs: map[string]string{
+				"field": "value",
+			},
+		},
+	}})
+	require.Nil(t, err)
+
+	m := &pb.Match{
+		MatchId:  "1",
+		Tickets:  []*pb.Ticket{t1},
+		Backfill: b,
+	}
+
+	om.SetMMF(func(ctx context.Context, profile *pb.MatchProfile, out chan<- *pb.Match) error {
+		out <- m
+		return nil
+	})
+
+	om.SetEvaluator(func(ctx context.Context, in <-chan *pb.Match, out chan<- string) error {
+		p, ok := <-in
+		require.True(t, ok)
+
+		out <- p.MatchId
+		return nil
+	})
+
+	stream, err := om.Backend().FetchMatches(ctx, &pb.FetchMatchesRequest{
+		Config:  om.MMFConfigGRPC(),
+		Profile: &pb.MatchProfile{},
+	})
+	require.Nil(t, err)
+	resp, err := stream.Recv()
+	require.Nil(t, err)
+	require.True(t, proto.Equal(m, resp.Match))
+
+	actual, err := om.Frontend().GetBackfill(ctx, &pb.GetBackfillRequest{BackfillId: b.Id})
+	require.Nil(t, err)
+	require.NotNil(t, actual)
+	require.True(t, proto.Equal(b, actual))
+
+	client, err := om.Query().QueryTickets(ctx, &pb.QueryTicketsRequest{Pool: &pb.Pool{
+		StringEqualsFilters: []*pb.StringEqualsFilter{{StringArg: "field", Value: "value"}},
+	}})
+
```

Comment:
We're also going to want to include making sure the backfill updates in query.  (having queried before so we know it has the old cached version.)  This isn't going to work until several pieces are in place though... just make sure it's on the work items that get done.  (either your tracker, or just an issue on github)

Reply:
You are talking about querying tickets or backfills? Using such code I just want to ensure that ticket associated with backfill will disappear from query results.

Reply:
Oh actually this just made me realize, when a backfill is rejected because of an update from the game server - we need to release the tickets because they were already marked as pending by the synchronizer.  (or move the marking as pending to the backend after verifying the backfill would work too.)

I was talking about the backfill having a new field value set by the mmf.

Reply:
The work re: unpending tickets because a backfill changed, can be done in a followup PR.